packages/wonder-blocks-data/src/util/__tests__/response-cache.test.js

```diff
@@ -436,105 +459,371 @@ describe("../response-cache.js", () => {
+                // Assert
+                expect(result).toBeFalsy();
+            });
+
+            describe("when server-side", () => {
+                beforeEach(() => {
+                    jest.spyOn(Server, "isServerSide").mockReturnValue(true);
+                });
+
+                it("should return false if nothing was removed from ssr-only cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    const ssrOnlyCache = new MemoryCache();
+                    jest.spyOn(hydrationCache, "remove").mockReturnValue(true);
+                    jest.spyOn(ssrOnlyCache, "remove").mockReturnValue(false);
+                    const cache = new ResponseCache(
+                        hydrationCache,
+                        ssrOnlyCache,
+                    );
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: false,
+                    };
+
+                    // Act
+                    const result = cache.remove(fakeHandler, "optionsA");
+
+                    // Assert
+                    expect(result).toBeFalsy();
+                });
+
+                it("should return true if something was removed from ssr-only cache", () => {
+                    // Arrange
+                    const hydrationCache = new MemoryCache();
+                    const ssrOnlyCache = new MemoryCache();
+                    jest.spyOn(ssrOnlyCache, "remove").mockReturnValue(true);
+                    const cache = new ResponseCache(
+                        hydrationCache,
+                        ssrOnlyCache,
+                    );
+                    const fakeHandler: IRequestHandler<string, string> = {
+                        getKey: () => "MY_KEY",
+                        type: "MY_HANDLER",
+                        fulfillRequest: jest.fn(),
+                        hydrate: false,
+                    };
+
+                    // Act
+                    const result = cache.remove(fakeHandler, "optionsA");
+
+                    // Assert
+                    expect(result).toBeTruthy();
+                });
+            });
         });
     });
 
     describe("#removeAll", () => {
-        it("should return total number of entries removed", () => {
-            // Arrange
-            const hydrationCache = new MemoryCache();
-            jest.spyOn(hydrationCache, "removeAll").mockReturnValue(1);
-            const cache = new ResponseCache(hydrationCache);
-            const fakeHandler: IRequestHandler<string, string> = {
-                getKey: () => "MY_KEY",
-                type: "MY_HANDLER",
-                fulfillRequest: jest.fn(),
-                hydrate: true,
-            };
+        describe("when handler wants hydration", () => {
```

Comment:
Does this only happen on the client?

Reply:
I'm not sure I understand the question - can you elaborate?

Reply:
Does the scenario of the handling "wanting hydration" only occur on the client or would this test fail if we were on the server?

Reply:
We likely wouldn't call this function on the server, but if we did, it would work just fine in both places.